### PR TITLE
Move over to new Azure SDK

### PR DESCRIPTION
**Pull request by tzhanl**
**Purpose**
---------------------------------------------------------------
Move over this sample to new Azure SDK:

1. Move the original file to storage-blobs-java-v10-quickstart-v3 folder
2. Add folder storage-blobs-java-v10-quickstart-v4 for the sample referenced to the new SDK
3. Add two sections to the readme:
   - This Quickstart shows how to do some basic operations of Storage Blobs
   - Azure SDK versions

**What old sample do:**
---------------------------------------------------------------
image

**What can be updated to V4:**
---------------------------------------------------------------
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,50 +1,40 @@
...
-```
-
-Windows
-```
-setx AZURE_STORAGE_ACCOUNT "<youracountname>"
-setx AZURE_STORAGE_ACCESS_KEY "<youraccountkey>"
-```
-
-At this point, you can run this application using maven: `mvn compile exec:java`. It creates its own file to upload and download, and then cleans up after itself by deleting everything at the end. 
+# Quick Start with Azure Storage SDK V10 for Java
```
Comment (elraikhm) [inline README.md]: Should it be a Blob Storage ?

File: README.md [hunk 2]
```diff
@@ -1,50 +1,40 @@
...
+
+> * Create a storage account using the Azure portal.
+> * Create a container.
+> * Upload a file to block blob.
+> * List blobs.
+> * Download a blob to file.
+> * Delete a blob.
+> * Delete the container.
+
+# Folders introduction
```
Comment (elraikhm) [inline README.md]: Not sure why do we call this "folders introduction"? Suggest to change to something like: 

Azure SDK versions
To use the latest Azure SDK version [storage-blobs-java-v10-quickstart-v4] please add the following dependency to the  Maven pom.xml file:
<>
For the previous stable Azure SDK version[storage-blobs-java-v10-quickstart-v3] please add the following dependencies to the  Maven pom.xml file..
<>

File: storage-blobs-java-v10-quickstart-v4/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Quick Start with Azure Storage SDK V10 for Java
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/README.md]: All my comments on v3 apply here.  Merge them into the same file.

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+public class Quickstart {
+    static File createTempFile() throws IOException {
+
+        // Here we are creating a temporary file to use for download and upload to Blob storage
+        File sampleFile = null;
+        sampleFile = File.createTempFile("sampleFile", ".txt");
+        System.out.println(">> Creating a sample file at: " + sampleFile.toString());
+        Writer output = new BufferedWriter(new FileWriter(sampleFile));
+        output.write("Hello Azure Storage blob quickstart!");
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: Remove all "!". Just use "."

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        output.close();
+
+        return sampleFile;
+    }
+
+    public static void main(String[] args) throws IOException {
+        ContainerClient containerClient;
+        String endpoint = "<-Primary Blob Service Endpoint->";
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: Are you asking the user to replace or should this come from env variables?  Also, placeholder text needs to be consistent format across all samples.

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        String containerName = SdkContext.randomResourceName("quickstart",20);
+        String blobName = "<-upload to file->";
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: In the other version of this we provide the blob name, why not here too?

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        String downloadFileName = SdkContext.randomResourceName("downloadFile", 25);
+        // Creating a sample file to use in the sample
+        File sampleFile = null;
+
+        sampleFile = createTempFile();
+
+        String downloadedFilePath = "<-download file path using downloadFileName (this file must not be exist)->";
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: Are you asking the user to change this?  I don't see where you are doing so.  

As a rule, they should only have to change env vars and not actually change the code at all to run it.

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        BlobClient blobClient = new BlobClientBuilder()
+        .endpoint(endpoint)
+        .credential(credential)
+        .containerName(containerName)
+        .blobName(blobName)
+        .buildBlobClient();
+ 
+        // Listening for commands from the console
+        System.out.println("Enter a command");
+        System.out.println("(P)utBlob-upload | (L)istBlobs | (G)etBlob-download | (D)eleteBlobs | (E)xitSample");
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: This is hard to read with Pascal and dash cases.  Change to:

"(U) Upload Blob | (L) List Blobs | (G) Get Blob | (D) Delete Blobs | (X) Exit "

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+        while (true) {
+
+            System.out.println("# Enter a command : ");
+            String input = reader.readLine();
+
+            switch (input) {
+
+            // Upload a blob from a File
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: The cases should be indented from the switch statement.

File: storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+            case "P":
```
Comment (jongio) [inline storage-blobs-java-v10-quickstart-v4/src/main/java/quickstart/Quickstart.java]: Change based on message change above.  This goes for all key shortcuts that have changed.

File: README.md [hunk 11]
```diff
@@ -1,29 +1,30 @@
-# Quick Start with Azure Storage SDK V10 for Java
-## How to run this project
+# Quickstart with Azure Storage Blobs SDK V10 for Java
 
 ### Prerequisites
 
-If you don't have an Azure subscription, create a [free account](https://azure.microsoft.com/free/?WT.mc_id=A261C142F) before you begin.
+If you don't have an Azure subscription, create a [Storage Account](https://azure.microsoft.com/free/?WT.mc_id=A261C142F) before you begin.
```
Comment (jongio) [inline README.md]: This should be 'free account', not 'Storage Account'

File: README.md [hunk 12]
```diff
@@ -45,9 +46,43 @@ At this point, you can run this application using maven: `mvn compile exec:java`
...
+
+> * Create a Storage Account using the Azure portal.
+> * Create a container.
+> * Upload a file to block blob.
+> * List blobs.
+> * Download a blob to file.
+> * Delete a blob.
+> * Delete the container.
+
+# Azure SDK versions
```
Comment (jongio) [inline README.md]: Please put "latest" first and "previous" after.

File: README.md [hunk 13]
```diff
@@ -45,9 +44,44 @@ At this point, you can run this application using maven: `mvn compile exec:java`
...
+     <artifactId>azure-storage-blob</artifactId>
+     <version>12.0.0</version>
+   </dependency>
+```
+* For the previous stable Azure SDK version [storage-blobs-java-v10-quickstart-v3] please add the following dependencies to the Maven pom.xml file :
+```xml
+   <dependency>
+     <groupId>com.microsoft.azure</groupId>
+     <artifactId>azure-storage-blob</artifactId>
+     <version>11.0.0</version>
```
Comment (alzimmermsft) [inline README.md]: While V11 has been the last stable release the Storage team has generally recommended against using it, instead should this list V8?

https://github.com/azure/azure-storage-java#we-will-continue-to-respond-to-issues-here-but-prefer-that-you-post-them-on-the-v12-repo-thank-you-for-your-patience-in-this-effort-we-look-forward-to-continuing-to-work-together-with-you

Comment (tzhanl) [reply README.md]: V8 API is totally different from V11. The V3 code need huge update to fit V8 package. Is it worth to do it?

File: README.md [hunk 14]
```diff
@@ -1,29 +1,28 @@
...
-First, create a new general-purpose storage account to use for this tutorial. 
+Step 1 : Create a new general-purpose storage account to use for this tutorial. 
 
-*  Go to the [Azure portal](https://portal.azure.com) and log in using your Azure account. 
-*  On the Hub menu, select **New** > **Storage** > **Storage account - blob, file, table, queue**. 
+*  Go to the [Azure Portal](https://portal.azure.com) and log in using your Azure account. 
+*  Select **New** > **Storage** > **Storage account**. 
+*  Select your Subscription. 
+*  For `Resource group`, create a new one and give it a unique name. 
 *  Enter a name for your storage account. The name must be between 3 and 24 characters in length and may contain numbers and lowercase letters only. It must also be unique.
```
Comment (jongio) [inline README.md]: "The name..." doesn't need to be in this doc.  It is already in the portal and may change.  If it changes, then we won't know and this text and that text will be out of sync.

File: README.md [hunk 15]
```diff
@@ -45,9 +44,44 @@ At this point, you can run this application using maven: `mvn compile exec:java`
 mvn compile exec:java
 ```
 
+### This Quickstart shows how to do some basic operations of Storage Blobs.
```
Comment (jongio) [inline README.md]: Change "some basic" to "the following"

File: README.md (hunk 13 above)
Comment (jongio) [reply README.md]: If that what was in the original sample, then I recommend keeping as-is.

File: README.md [hunk 16]
```diff
@@ -1,29 +1,54 @@
-# Quick Start with Azure Storage SDK V10 for Java
-## How to run this project
+# Quickstart with Azure Storage Blobs SDK V10 for Java
+
+### SDK Versions
+You will find the following folders: storage-blobs-java-v10-quickstart-v11, which references the version 11.0.0 of the SDK and storage-blobs-java-v10-quickstart-v12, which uses the version 12.0.0 of the SDK.
```
Comment (jongio) [inline README.md]: Please see my format from the key-vault sample and use the same here.

File: README.md [hunk 17]
```diff
@@ -1,29 +1,54 @@
-# Quick Start with Azure Storage SDK V10 for Java
```
Comment (jongio) [inline README.md]: This file needs the yaml front matter, like all the other samples.

File: README.md [hunk 18]
```diff
@@ -1,29 +1,40 @@
...
-## How to run this project
+---
+page_type: sample
+languages: Java
```
Comment (jongio) [inline README.md]: These needs to match the "slug", not the name.  So all lower case, and double check all products across all samples.

**Pull request merged**